src/test/java/fi/vm/yti/datamodel/api/v2/service/JSONValidationServiceTest.java

```diff
@@ -0,0 +1,74 @@
+package fi.vm.yti.datamodel.api.v2.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertLinesMatch;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@ExtendWith(SpringExtension.class)
+
+@Import({
+    JSONValidationService.class
+})
+
+public class JSONValidationServiceTest {
+	
+	String metaSchemaPath = "schema_v4";
+	InputStream metaSchemaInputStream = getClass().getClassLoader().getResourceAsStream(metaSchemaPath);
+	
+	// have to restart the below one (comment - save - uncomment - save) each time the file is modified
+	String secondInvalidInputSchemaPath = "test_jsonschema_invalid_schema.json";
+	
+	
+	private byte[] byteStreamFromPath(String schemaPath) throws Exception, IOException {
+		InputStream inputSchemaInputStream = getClass().getClassLoader().getResourceAsStream(schemaPath);
+		byte[] inputSchemaInByte = inputSchemaInputStream.readAllBytes();
+		inputSchemaInputStream.close();
+		
+		return inputSchemaInByte;
+	}
+	
+	@Test
+	void testValidJSONSchema() throws Exception, IOException {
+		String validInputSchemaPath = "test_jsonschema_b2share.json";
+		// *** validate format (eg email, date etc) with regex? ***
+		
+		assertEquals(JSONValidationService.validateJSON(byteStreamFromPath(validInputSchemaPath)), new ValidationRecord(true, Arrays.asList()));
```

Comment:
validataJSON() method does not exist. Have you compiled the code? Does the tests run successfully?